smoke-test/tests/cypress/cypress/e2e/mutations/dataset_ownership.js

```diff
@@ -0,0 +1,106 @@
+const test_id = Math.floor(Math.random() * 100000);
+const username = `Example Name ${test_id}`;
+const email = `example${test_id}@example.com`;
+const password = "Example password";
+const group_name = `Test group ${test_id}`;
+
+const addOwner = (owner, type, elementId) => {
+    cy.clickOptionWithText("Add Owners");
+    cy.contains("Search for users or groups...").click({ force: true });
+    cy.focused().type(owner);
+    cy.clickOptionWithText(owner);
+    cy.focused().blur();
+    cy.waitTextVisible(owner);
+    cy.get('[role="dialog"]').contains("Technical Owner").click();
+    cy.get('[role="listbox"]').parent().contains(type).click();
+    cy.get('[role="dialog"]').contains(type).should("be.visible");
+    cy.clickOptionWithText("Done");
+    cy.waitTextVisible("Owners Added");
+    cy.waitTextVisible(type);
+    cy.waitTextVisible(owner).wait(3000);
+    cy.clickOptionWithText(owner);
+    cy.waitTextVisible("SampleCypressHiveDataset");
+    cy.goToDataset("urn:li:dataset:(urn:li:dataPlatform:hive,SampleCypressHiveDataset,PROD)", "SampleCypressHiveDataset");
+    cy.get(elementId).next().click();
+    cy.clickOptionWithText("Yes");
+    cy.waitTextVisible("Owner Removed");
+    cy.ensureTextNotPresent(owner);
+    cy.ensureTextNotPresent(type);
+}
+
+describe("add, remove ownership for dataset", () => {
+    it("create test user and test group, add user to a group", () => {
+        cy.loginWithCredentials();
+        cy.visit("/settings/identities/users");
+        cy.waitTextVisible("Invite Users");
+        cy.clickOptionWithText("Invite Users");
+         cy.waitTextVisible(/signup\?invite_token=\w{32}/).then(($elem) => {
```

Comment:
I almost think we should make this into a repeatable method :)

Btw, indenting look a bit odd here

Reply:
(This could almost be a createNewUser or signUpNewUser flow

Reply:
I think we can create: createUser, createGroup and addGroupUser methods for future use.

Reply:
In this case I've used existing flow for user creation from add_users.js